launch_xml/launch_xml/entity.py

```diff
@@ -53,8 +55,23 @@ def parent(self) -> Optional['Entity']:
     @property
     def children(self) -> List['Entity']:
         """Get the Entity's children."""
+        self.__read_children = {item.tag for item in self.__xml_element}
         return [Entity(item) for item in self.__xml_element]
 
+    def assert_entity_complely_parsed(self):
+        unparsed_nested_tags = {item.tag for item in self.__xml_element} - self.__read_children
```

Comment:
@ivanpauno shouldn't we assert recursively? I'm thinking of nested structures that are parsed by the same parsing function e.g. nested `launch_ros.actions.Node` parameters. Thinking out loud, perhaps we can keep some form of context in the parser to avoid full traversals at each node of the tree, but that's arguably early optimization.

Reply:
> shouldn't we assert recursively?

That doesn't sound like a good idea to me. I mean for things like `group` or `timer` it shouldn't be recursive, but for `executable` or `node` that could work.

>  I'm thinking of nested structures that are parsed by the same parsing function e.g. nested launch_ros.actions.Node parameters

Yeah that's true, maybe in those places we need to call `assert_entity_completely_parsed` manually (as `param`, `env`, etc don't have an explicit parsing function that can be wrapped).

Reply:
See https://github.com/ros2/launch/pull/468/commits/caf4b0cf338ee55218cc3c737d17c58d861f17af

Reply:
and https://github.com/ros2/launch_ros/pull/198